### PR TITLE
UCT/SELF: Cleanups in self transport

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6310/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6311/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6314/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6315/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg pls take a look

File: src/uct/sm/self/self.c [hunk 1]
```diff
@@ -0,0 +1,363 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
```
Comment (hoopoepg) [inline src/uct/sm/self/self.c]: s/2016/2018

File: src/uct/sm/self/self.c [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    const uct_self_iface_t *iface = ucs_derived_of(tl_iface, uct_self_iface_t);
+
+    *(uct_self_iface_addr_t*)addr = iface->id;
+    return UCS_OK;
+}
+
+static int uct_self_iface_is_reachable(const uct_iface_h iface, const uct_device_addr_t *dev_addr,
+                                       const uct_iface_addr_t *iface_addr)
+{
+    const uct_self_iface_t *self_iface = NULL;
```
Comment (hoopoepg) [inline src/uct/sm/self/self.c]: I see that code is just copied from another file, but why we initialize vars here? 3 lanes below we re-init it...
and alignment is not ok

File: src/uct/sm/self/self.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
```
Comment (hoopoepg) [inline src/uct/sm/self/self.h]: s/2016/2018

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6316/ for details (Mellanox internal link).

**Pull request merged**